storage/replica_raftstorage.go

```diff
@@ -287,12 +293,14 @@ func (r *Replica) Snapshot() (raftpb.Snapshot, error) {
 		// state of the Replica). Everything must come from the snapshot.
 		snapData, err := snapshot(snap, rangeID, r.mu.state.Desc.StartKey)
 		if err != nil {
-			log.Errorf(context.TODO(), "%s: error generating snapshot: %s", r, err)
+			log.Errorf(context.Background(), "%s: error generating snapshot: %s", r, err)
```

Comment:
We can't thread a real context into this method because it's an external API, so is there any reason to extract a local variable for context.Background()? (and there's an async task here so I think we'd need at least two contexts)